File: 900/2110B/main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() 
{
    int t;
    cin >> t;

    while (t--) 
    {
        string s;
        cin >> s;

        string ans = "NO";
        int o = 0;

        for (int i = 0; i < (int)s.size(); i++) 
        {
            if (s[i] == '(')
                o++;
            else
                o--;

            if (o == 0 && i != (int)s.size() - 1)
                ans = "YES";
        }

        cout << ans << "\n";
    }

    return 0;
}

File: 900/2110B/main.java
import java.util.*;
public class main
{
    public static void main(String args[])
    {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();

        while(t-- > 0)
        {
            String ans = "NO";

            String s = sc.next();
            int o = 0;

            for(int i=0;i<s.length();i++)
            {
                if(s.charAt(i) == '(')
                    o++;
                else
                    o--;
                
                if(o == 0 && i != s.length()-1)
                    ans = "YES";
            }

            System.out.println(ans);
        }
    }
}